### ISSUE TITLE
[Feature] optimize performance

### ISSUE THREAD

**Issue by GrinZero** [goto-bus-stop/node-resolve#13]
Hello 👋 This is a very good Rust package that solves my needs very well, but when I analyzed the program speed using a flame graph, I found that 'node resolve' took up too much time in it.
Do you want to confirm if there is a performance optimization plan?

**Comment by goto-bus-stop** [comment]
Depending on what else you're doing, it might be expected that this crate is a bit slow because it does a lot of synchronous I/O.

**Comment by GrinZero** [comment]
Indeed, although I made some magical modifications based on the code. Added some caching mechanisms, but the core issue remains unresolved.
